app/lib/proxy_config_affecting_changes.rb

```diff
@@ -1,115 +1,158 @@
-  module ProxyRuleExtension
-    extend ActiveSupport::Concern
+  class TrackedObject
+    def initialize(object)
+      @object = object
+      @initial_state = object.proxy_config_affecting_state.hash
+      @new_record = object.new_record?
+      @proxies = find_proxies(object)
+    end
 
-    included do
-      include ProxyConfigAffectingChanges
+    attr_reader :object, :initial_state, :new_record
 
-      after_commit :issue_proxy_affecting_change_events
+    def ==(other)
+      object == other.object
+    end
 
-      def issue_proxy_affecting_change_events
-        return unless owner
+    delegate :destroyed?, :persisted?, to: :object
 
-        case owner
-        when Proxy
-          issue_proxy_affecting_change_event(owner)
-        when BackendApi
-          owner.proxies.find_each(&method(:issue_proxy_affecting_change_event))
-        end
-      end
+    def created?
+      new_record && persisted?
+    end
+
+    def state_changed?
+      initial_state != object.reload.proxy_config_affecting_state.hash
+    rescue ActiveRecord::RecordNotFound
+      !new_record # Cannot find the model. If the object was a new record, then it was not persisted; otherwise, it was destroyed.
+    end
+
+    def changed?
+      destroyed? || created? || state_changed?
+    end
+
+    def proxies
+      @proxies.presence || find_proxies
+    end
+
+    protected
+
+    def find_object
+      object.class.find_by(id: object.id)
+    end
+
+    def find_proxies(model = find_object)
+      proxy_or_proxies = model.try(:proxy) || model.try(:proxies)
+      [*proxy_or_proxies].flatten.compact
     end
   end
 
-  module ProxyExtension
-    extend ActiveSupport::Concern
+  class Tracker
+    def initialize
+      @tracked_objects = []
+    end
 
-    included do
-      has_one :proxy_config_affecting_change, dependent: :delete
-      private :proxy_config_affecting_change
+    def track(object)
+      return unless object
+      tracked_object = TrackedObject.new(object)
+      return if tracking?(tracked_object)
+      @tracked_objects << tracked_object
+    end
 
-      include ProxyConfigAffectingChanges
+    def tracking?(object)
+      tracked_object = object.is_a?(TrackedObject) ? object : TrackedObject.new(object)
+      @tracked_objects.include?(tracked_object)
+    end
 
-      after_commit :issue_proxy_affecting_change_events, on: :update
+    def objects_with_affecting_changes
+      @tracked_objects.select(&:changed?)
+    end
 
-      def issue_proxy_affecting_change_events
-        return if previously_changed?(:created_at) || (previous_changes.keys - %w[updated_at lock_version]).empty?
+    def flush
+      proxies = objects_with_affecting_changes.map(&:proxies).flatten.uniq
+      proxies.each(&method(:issue_proxy_affecting_change_event))
+    ensure
+      @tracked_objects.clear
+    end
 
-        issue_proxy_affecting_change_event(self)
-      end
+    # FIXME: This is only so ProxyConfigs::AffectingObjectChangedEvent does not crash
```

Comment:
Could you describe the issue so we could help a bit?

Reply:
https://github.com/3scale/porta/blob/4b618c0fe830d85a245a53a9d1f2c7ddb0725c3d/app/events/proxy_configs/affecting_object_changed_event.rb#L7